cottontaildb_client/cottontaildb_client.py

```diff
@@ -343,6 +344,41 @@ def ping(self):
         """Sends a ping message to the endpoint. If method returns without exception endpoint is connected."""
         self._dql.Ping(Empty())
 
+    def nns(self, schema, entity, query_vector, distance='manhattan', limit=None, vector_col='feature', id_col='id'):
+        """
+        Queries the specified entity with the given vector.
+
+        @param schema: the schema containing the queried entity
+        @param entity: the entity being queried
+        @param query_vector: the query vector. Simple float array.
+        @param distance: the distance to be used
+        @param limit: maximum number of rows to return
+        @param vector_col: column name where the vector is stored
+        @param id_col: column name where the id is stored
+        """
+        schema_name = SchemaName(name=schema)
+        entity_name = EntityName(schema=schema_name, name=entity)
+        nnscol = Expression(column=ColumnName(entity=entity_name, name=vector_col))
+
+        distancecol = ColumnName(name='distance')
+        idExpression = Expression(column=ColumnName(name='id'))
+
+        nnslit = Expression(literal=float_vector(*query_vector))
+        fn = FunctionName(name='euclidean')
```

Comment:
Is it possible the caller provided distance function is never used?
I can't find any reference to it and `euclidean` appears hard-coded.